engine/language-server/src/main/scala/org/enso/languageserver/vcsmanager/Git.scala

```diff
@@ -45,17 +51,45 @@ private class Git extends VcsApi[BlockingIO] {
       if (!rootFile.exists()) {
         throw new FileNotFoundException("unable to find project repo: " + root)
       }
-      val repoLocation = root.resolve(DefaultGitRepoDir)
+
+      val repoLocation = root.resolve(gitDir)
       if (repoLocation.toFile.exists()) {
         throw new RepoExists()
       }
 
+      if (!repoLocation.getParent.toFile.exists()) {
+        if (!repoLocation.getParent.toFile.mkdirs()) {
+          throw new FileNotFoundException(
+            "unable to create project repository at " + repoLocation
+          )
+        }
+      }
+
       val jgit = JGit
         .init()
-        .setDirectory(rootFile)
+        .setGitDir(repoLocation.toFile)
+        .setDirectory(root.toFile)
         .setBare(false)
         .call()
 
+      ensoDataDirectory.foreach { path =>
+        // JGit **always** creates a .git file, gitdir path.
+        // The file may be confusing to the users and unnecessary for
+        // VCS since all commands include repo path explicitly.
+        val dotGit = root.resolve(".git").toFile
+        if (dotGit.exists()) {
+          dotGit.delete()
+        }
+
+        if (path.toFile.exists()) {
+          throw new EntryExistsException(path.toString)
+        }
+        Files.write(
+          root.resolve(GitIgnore),
+          path.toString.getBytes(StandardCharsets.UTF_8)
+        )
```

Comment:
0. it will overwrite user's `.gitignore` file if it exists
1. it assumes that users will use git. Otherwise it just creates more unnecessary files in the root directory
2. if the user initialized a git repo in the project root, they should have already noticed the `.enso` directory and added it to ignored files

IMO users should handle `.enso` directory themselves

Reply:
I would tend to agree except that `.git` is itself in `.enso`. This makes is pretty problematic because every time you commit, you still get dirty status as `.git` change. catch-22.

I will see if I can do some trickery to avoid .gitignore, it is not the prettiest hack, I agree.